Testscripts/Linux/LSVMBUS.sh

```diff
@@ -100,45 +100,46 @@ if ! which bc; then
     update_repos
     install_package bc
 fi
+TIMESTAMP=$(date +%Y%m%d-%H%M)
+$lsvmbus_path -vvv > "lsvmbus_$TIMESTAMP.log"
+net_adapter_index=0
+network_counter=0
+scsi_index=0
+scsi_counter=0
 
-$lsvmbus_path -vvv > lsvmbus.log
-
-# Check number of NICs on VM
-nics=$( grep -o "Synthetic network adapter" lsvmbus.log | wc -l)
-if [ "$nics" -gt 1 ]; then
-    LogMsg "Counting the cores spread only for the first NIC..."
-    sed -i ':a;N;$!ba;s/Synthetic network adapter/ignored adapter/2' lsvmbus.log && \
-    sed -i '/ignored adapter/,/^$/d' lsvmbus.log
-fi
-
-# Check number of SCSI Controllers on VM
-scsiAdapters=$( grep -o "Synthetic SCSI Controller" lsvmbus.log | wc -l)
-if [ "$scsiAdapters" -gt 1 ]; then
-    LogMsg "Counting the cores spread only for the first SCSI Adapter..."
-    sed -i ':a;N;$!ba;s/Synthetic SCSI Controller/ignored controller/2' lsvmbus.log && \
-    sed -i '/ignored controller/,/^$/d' lsvmbus.log
-fi
+declare -A ADAPTER_DICT=()
+declare -A SCSI_DICT=()
 
 while IFS='' read -r line || [[ -n "$line" ]]; do
-    if [[ $line =~ "VMBUS ID" ]]; then
+    if [[ "$line" =~ "VMBUS ID" ]]; then
         token=""
     fi
-    if [[ $line =~ "Synthetic network adapter" ]]; then
+
+    if [[ "$line" =~ "Synthetic SCSI Controller" ]]; then
+        ((scsi_index++))
+        token="controller"
+    fi
+
+    if [[ "$line" =~ "Synthetic network adapter" ]]; then
+        ((net_adapter_index++))
         token="adapter"
     fi
 
-    if [[ $line =~ "Synthetic SCSI Controller" ]]; then
-        token="controller"
+    if [[ "$line" == "" ]]; then
+        token=""
+        network_counter=0
+        scsi_counter=0
+    fi
+
+    if [[ "$token" == "adapter" ]] && [[ "$line" =~ "target_cpu" ]]; then
+        ADAPTER_DICT["adapter$net_adapter_index"]=$((++network_counter))
     fi
 
-    if [[ -n $token ]] && [[ $line =~ "target_cpu" ]]; then
-        if [[ $token == "adapter" ]]; then
-            ((network_counter++))
-        elif [[ $token == "controller" ]]; then
-            ((scsi_counter++))
-        fi
+    if [[ "$token" == "controller" ]] && [[ "$line" =~ "target_cpu" ]]; then
```

Comment:
You need to prepare the exceptional case as $token got set in the previous iteration, but if the $line has something else, then your looping will fail. like $line = "I am new string" Maybe in the beginning of while loop, reset $token value?

Reply:
I don't understand.